### DISCUSSION TITLE
Getting dynamic pathname including the query in Next.js 13

### DISCUSSION THREAD

**Discussion by JasperAlexander** [vercel/next.js#45623] [Help]
How to get the current pathname with dynamic segments, including the query, like asPath in Next.js 12? For example, when I'm on app/[username]/page.tsx, I want to get the pathname as /[username], not with the username itself.

**Comment by ciruz** [comment]
Hey,

it's possible with a combination of ``page`` params and ``usePathname()`` to get the information you want. That's the only way I'm currently aware of.

A simple example ``page.jsx`` placed in ``blog/page.jsx``, ``blog/[slug]/page.jsx`` and ``blog/foobar/page.jsx``

```javascript
"use client";

import { usePathname } from "next/navigation";

export default function Page({ params, searchParams }) {
  const pathname = usePathname();

  return (
    <>
      <b>Current pathname: </b>
      {pathname}
      <br />
      <b>Current params:</b>
      {Object.entries(params).map(([key, value]) => (
        <div key={key}>
          {key}: {value}
          <br />
        </div>
      ))}
    </>
  );
}
```


file: ``blog/page.jsx``
url: http://localhost:3000/blog/
result:
```
Current pathname: /blog
Current params:
```

file: ``blog/[slug]/page.jsx``
url: http://localhost:3000/blog/test
result:
```
Current pathname: /blog/test
Current params:
slug: test
```
(slug param is now set)

file: ``blog/foobar/page.jsx``
url: http://localhost:3000/blog/foobar
result
```
Current pathname: /blog/foobar
Current params:
```
